pkg/controllers/disruption/orchestration/queue.go

```diff
@@ -181,24 +193,19 @@ func NewCommand(replacements []string, candidates []*state.StateNode, id types.U
 func (q *Queue) Register(_ context.Context, m manager.Manager) error {
 	return controllerruntime.NewControllerManagedBy(m).
 		Named("disruption.queue").
-		WatchesRawSource(singleton.Source()).
-		Complete(singleton.AsReconciler(q))
+		WatchesRawSource(source.Channel(q.source, &handler.TypedEnqueueRequestForObject[*v1.NodeClaim]{})).
+		WithOptions(controller.Options{
+			RateLimiter:             reasonable.RateLimiter(),
+			MaxConcurrentReconciles: 100,
+		}).
+		Complete(reconcile.AsReconciler(m.GetClient(), q))
 }
 
-func (q *Queue) Reconcile(ctx context.Context) (reconcile.Result, error) {
+func (q *Queue) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
 	ctx = injection.WithControllerName(ctx, "disruption.queue")
-
-	// Check if the queue is empty. client-go recommends not using this function to gate the subsequent
-	// get call, but since we're popping items off the queue synchronously retrying, there should be
-	// no synchonization issues.
-	if q.Len() == 0 {
-		return reconcile.Result{RequeueAfter: 1 * time.Second}, nil
-	}
-
-	// Get command from queue. This waits until queue is non-empty.
-	cmd, shutdown := q.TypedRateLimitingInterface.Get()
-	if shutdown {
-		panic("unexpected failure, disruption queue has shut down")
+	cmd, exists := q.providerIDToCommand[nodeClaim.Status.ProviderID]
+	if !exists {
+		return reconcile.Result{}, fmt.Errorf("no command found for nodeclaim %s", nodeClaim.Name)
```

Comment:
This shouldn't happen but this is going to get into a continuous loop if we return an error here -- seems fine to just throw out the command if this happens and say that this is a no-op if the command isn't properly stored